### PR TITLE
fixes for coupled model P7 restart tests and Update build system and regression testing on Acorn (PR#809)

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This PR contains the NoahMP and CA fixes for coupled tests restart reproducibility. It also adds a standalone FV3 P7 CCPP suite file. It does not change the current regression test results. 

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes issue in discussion [#797](https://github.com/ufs-community/ufs-weather-model/discussions/797)

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

- ufs-weather-model [PR#819](https://github.com/ufs-community/ufs-weather-model/pull/819)
- fv3atm [PR#391](https://github.com/NOAA-EMC/fv3atm/pull/391)
- CCPP physics [PR#732](https://github.com/NCAR/ccpp-physics/pull/732)
- Stochastic physics [PR#46](https://github.com/noaa-psd/stochastic_physics/pull/46)

### CODE REVIEW

**Review by DeniseWorthen — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/739232976/20210923180008/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_32526
Test cpld_control_wave 015 failed failed 
Test cpld_control_wave 015 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by junwang-noaa** [comment]
> 
> 
> _**Automated RT Failure Notification**_
> Machine: herav
> Compiler: intel
> Job: RT
> Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/739232976/20210923180008/ufs-weather-model
> Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_32526
> Test cpld_control_wave 015 failed failed
> Test cpld_control_wave 015 failed in run_test failed
> Please make changes and add the following label back:
> hera-intel-RT

The error message from cpld_control_wave, the job did not start, will rerun this job
+ srun --label -n 400 ./fv3.exe
srun: error: slurm_receive_msgs: Socket timed out on send/recv operation
srun: error: Task launch for StepId=23466032.0 failed on node h17c43: Socket timed out on send/recv operation
srun: error: Application launch failed: Socket timed out on send/recv operation
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
  0: slurmstepd: error: *** STEP 23466032.0 ON h3c54 CANCELLED AT 2021-09-23T18:17:23 ***
srun: error: h3c54: tasks 0-39: Killed
srun: launch/slurm: _step_signal: Terminating StepId=23466032.0
srun: error: h4c50: tasks 40-79: Killed

**Comment by climbfuji** [comment]
Yay, gaea is back like in the good old days. The runtimes are reasonable again.

File: tests/abort_dep_tasks.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 from __future__ import print_function
-import ecflow
+import ecflow as ecflow
```
Comment (climbfuji) [inline tests/abort_dep_tasks.py:3]: Why is this required?

Comment (DusanJovic-NOAA) [reply tests/abort_dep_tasks.py:3]: I was getting some errors on Acorn, maybe something was wrong with ecflow installation. They keep changing things, I'll test again and remove it if it's not necessary anymore.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]

File: tests/abort_dep_tasks.py (hunk 1 above)
Comment (DusanJovic-NOAA) [reply tests/abort_dep_tasks.py:3]: Looks like they installed the latest version of ecflow the same day, or day before I made this change:
```
$ ls -dl /apps/ops/prod/nco/core/ecflow.v5.6.0.*
drwxr-xr-x 8 ops.prod prod 87 Apr 14 00:08 /apps/ops/prod/nco/core/ecflow.v5.6.0.3
drwxr-xr-x 8 ops.prod prod 87 Aug  5 16:15 /apps/ops/prod/nco/core/ecflow.v5.6.0.4
drwxr-xr-x 8 ops.prod prod 87 Aug 17 15:39 /apps/ops/prod/nco/core/ecflow.v5.6.0.5
drwxr-xr-x 8 ops.prod prod 87 Sep  2 19:58 /apps/ops/prod/nco/core/ecflow.v5.6.0.6
```
Who knows what the differences are between 5.6.0.5 and 5.6.0.6.
